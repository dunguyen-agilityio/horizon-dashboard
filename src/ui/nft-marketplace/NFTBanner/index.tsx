// Components
import Link from 'next/link';
import { Text } from '@/components';
import ImageWithBlur from '@/components/ImageWithBlur';

// Constants
import { NFT_IMAGES } from '@/constants/images';

// Types
import { TEXT_SIZE } from '@/types/text';

const { COVER } = NFT_IMAGES;

const NFTBanner = () => (
  <div className="relative h-[420px] rounded-md bg-blue-450 dark:bg-purple-750 overflow-hidden">
    <div className="absolute right-0 h-[350px] w-full max-w-[1090px]">
      <ImageWithBlur
        src={COVER}
        alt="NFT Cover"
        fill
        additionalClasses="object-cover"
        sizes="(max-width: 440px) 60vw, 100vw"
      />
    </div>
    <div className="absolute inset-0 z-10 px-5 p-5 md:p-10">
      <Text
        as="h1"
        size={TEXT_SIZE['2xl']}
        className="text-white whitespace-break-spaces"
      >
        {'Discover, collect, and\nsell extraordinary NFTs'}
      </Text>
      <Text
        className="text-purple-150 mt-[14px] leading-medium whitespace-break-spaces"
        size={TEXT_SIZE.md}
      >
        {
          'Enter in this creative world. Discover now the\nlatest NFTs or start creating your own!'
        }
      </Text>
      <Link
        className="text-purple-150 text-medium block w-fit mt-[41px] hover:underline font-medium"
        href="#"
      >
        Watch video
      </Link>
    </div>
  </div>
);

export default NFTBanner;
