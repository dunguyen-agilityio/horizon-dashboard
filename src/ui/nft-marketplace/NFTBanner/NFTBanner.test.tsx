import { render } from '@testing-library/react';

import NFTBanner from '.';

describe('NFTBanner tests', () => {
  it('Should match snapshot', () => {
    const { container } = render(<NFTBanner />);
    expect(container).toMatchSnapshot();
  });
});
