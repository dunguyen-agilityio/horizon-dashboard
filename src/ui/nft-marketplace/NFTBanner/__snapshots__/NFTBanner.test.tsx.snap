// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFTBanner tests Should match snapshot 1`] = `
<div>
  <div
    class="relative h-[420px] rounded-md bg-blue-450 dark:bg-purple-750"
  >
    <div
      class="absolute right-0 h-[350px] w-full max-w-[1090px]"
    >
      <img
        alt="NFT Cover"
        class="object-cover"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 440px) 60vw, 100vw"
        src="/_next/image?url=%2Fnft-cover.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fnft-cover.webp&w=384&q=75 384w, /_next/image?url=%2Fnft-cover.webp&w=640&q=75 640w, /_next/image?url=%2Fnft-cover.webp&w=750&q=75 750w, /_next/image?url=%2Fnft-cover.webp&w=828&q=75 828w, /_next/image?url=%2Fnft-cover.webp&w=1080&q=75 1080w, /_next/image?url=%2Fnft-cover.webp&w=1200&q=75 1200w, /_next/image?url=%2Fnft-cover.webp&w=1920&q=75 1920w, /_next/image?url=%2Fnft-cover.webp&w=2048&q=75 2048w, /_next/image?url=%2Fnft-cover.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div
      class="absolute inset-0 z-10 px-5 p-5 md:p-10"
    >
      <h1
        class="text-primary dark:text-white text-2xl font-bold leading-2xl text-white whitespace-break-spaces"
      >
        Discover, collect, and
sell extraordinary NFTs
      </h1>
      <p
        class="text-primary dark:text-white text-medium text-purple-150 mt-[14px] leading-medium whitespace-break-spaces"
      >
        Enter in this creative world. Discover now the
latest NFTs or start creating your own!
      </p>
      <a
        class="text-purple-150 text-medium block w-fit mt-[41px] hover:underline font-medium"
        href="#"
      >
        Watch video
      </a>
    </div>
  </div>
</div>
`;
